audio/src/conn.rs

```diff
@@ -1,86 +1,545 @@
-use crate::{AudioMessage, Command, CommandsMessage, ExportState, Player, SynthState, TimeState};
-use crossbeam_channel::{Receiver, Sender};
+use crate::decayer::Decayer;
+use crate::export::{ExportState, ExportType, Exportable, MultiFileSuffix};
+use crate::exporter::Exporter;
+use crate::play_state::PlayState;
+use crate::types::SharedPlayState;
+use crate::SharedExportState;
+use crate::{
+    midi_event_queue::MidiEventQueue, types::SharedSample, Command, Player, Program,
+    SharedMidiEventQueue, SharedSynth, SynthState,
+};
+use common::open_file::Extension;
+use common::{MidiTrack, Music, PathsState, State, Time, MAX_VOLUME};
+use hashbrown::HashMap;
+use oxisynth::{MidiEvent, SoundFont, SoundFontId, Synth};
+use parking_lot::Mutex;
+use std::fs::File;
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+use std::thread::spawn;
+
+/// A convenient wrapper for a SoundFont.
+struct SoundFontBanks {
+    id: SoundFontId,
+    /// The banks and their presets.
+    banks: HashMap<u32, Vec<u8>>,
+}
+
+impl SoundFontBanks {
+    pub fn new(font: SoundFont, synth: &mut SharedSynth) -> Self {
+        let mut banks: HashMap<u32, Vec<u8>> = HashMap::new();
+        (0u32..129u32).for_each(|b| {
```

Comment:
Maybe this is more clear as

```suggestion
        (0u32..=128u32).for_each(|b| {
```